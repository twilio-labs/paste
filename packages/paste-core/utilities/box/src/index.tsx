import styled from '@emotion/styled';
import {
  display,
  DisplayProps,
  borders,
  BorderProps,
  height,
  maxWidth,
  minWidth,
  space,
  style,
  width,
} from 'styled-system';

import {
  SpacingThemeProps,
  BackgroundColorProp,
  BorderRadiusProps,
  BorderWidthProps,
  WidthProps,
  HeightProp,
  MaxWidthProp,
  MinWidthProp,
  BorderColorProps,
  BorderStyleProp,
} from './types';

export interface BoxProps
  extends DisplayProps,
    BorderProps,
    HeightProp,
    MaxWidthProp,
    MinWidthProp,
    WidthProps,
    SpacingThemeProps,
    BackgroundColorProp,
    BorderColorProps,
    BorderRadiusProps,
    BorderStyleProp,
    BorderWidthProps {
  as?: keyof JSX.IntrinsicElements;
}

const backgroundColor = style({
  prop: 'backgroundColor',
  cssProperty: 'backgroundColor',
  key: 'backgroundColors',
});

const borderBottomLeftRadius = style({
  prop: 'borderBottomLeftRadius',
  cssProperty: 'borderBottomLeftRadius',
  key: 'radii',
});

const borderBottomRightRadius = style({
  prop: 'borderBottomRightRadius',
  cssProperty: 'borderBottomRightRadius',
  key: 'radii',
});

const borderTopLeftRadius = style({
  prop: 'borderTopLeftRadius',
  cssProperty: 'borderTopLeftRadius',
  key: 'radii',
});

const borderTopRightRadius = style({
  prop: 'borderTopRightRadius',
  cssProperty: 'borderTopRightRadius',
  key: 'radii',
});

const borderColor = style({
  prop: 'borderColor',
  cssProperty: 'borderColor',
  key: 'borderColors',
});

const borderBottomWidth = style({
  prop: 'borderBottomWidth',
  cssProperty: 'borderBottomWidth',
  key: 'borderWidths',
});

const borderLeftWidth = style({
  prop: 'borderLeftWidth',
  cssProperty: 'borderLeftWidth',
  key: 'borderWidths',
});

const borderRightWidth = style({
  prop: 'borderRightWidth',
  cssProperty: 'borderRightWidth',
  key: 'borderWidths',
});

const borderTopWidth = style({
  prop: 'borderTopWidth',
  cssProperty: 'borderTopWidth',
  key: 'borderWidths',
});

const Box = styled.div<BoxProps>`
  ${backgroundColor};
  ${borders};
  ${borderColor}
  ${borderBottomWidth}
  ${borderLeftWidth}
  ${borderRightWidth}
  ${borderTopWidth}
  ${borderBottomLeftRadius};
  ${borderBottomRightRadius};
  ${borderTopLeftRadius};
  ${borderTopRightRadius};
  ${display};
  ${height};
  ${maxWidth};
  ${minWidth};
  ${space};
  ${width};
`;

Box.defaultProps = {
  minWidth: 'size0',
};

export {Box};
