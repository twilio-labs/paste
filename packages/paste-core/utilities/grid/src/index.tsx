export * from './Grid';
export * from './Column';
