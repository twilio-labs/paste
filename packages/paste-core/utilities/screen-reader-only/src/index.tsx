import styled from '@emotion/styled';

export const ScreenReaderOnly = styled.span<{}>`
  position: absolute;
  margin: -1px;
  border: 0;
  padding: 0;
  width: 1px;
  height: 1px;
  overflow: hidden;
  clip: rect(0 0 0 0);
  text-transform: none;
  white-space: nowrap;
`;
