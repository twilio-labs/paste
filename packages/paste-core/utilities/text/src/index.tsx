import {ReactNode} from 'react';
import styled from '@emotion/styled';
import {
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  space,
  SpaceProps,
  style,
  textAlign,
  TextAlignProps,
} from 'styled-system';
import {ThemeShape} from '@twilio-paste/theme-tokens';

interface TextProps extends TextAlignProps, SpaceProps {
  as?: string;
  children: ReactNode | string;
  fontFamily?: keyof ThemeShape['fonts'];
  fontSize?: keyof ThemeShape['fontSizes'];
  fontWeight?: keyof ThemeShape['fontWeights'];
  lineHeight?: keyof ThemeShape['lineHeights'];
  textColor?: keyof ThemeShape['textColors'];
  theme?: ThemeShape;
}

const textColor = style({
  prop: 'textColor',
  cssProperty: 'color',
  key: 'textColors',
});

const Text = styled.span<TextProps>`
  margin: 0;
  padding: 0;
  ${fontFamily}
  ${fontSize}
  ${textColor}
  ${fontWeight}
  ${lineHeight}
  ${textAlign}
  ${space}
`;

Text.defaultProps = {
  as: 'p',
  fontSize: 'fontSize30',
  lineHeight: 'lineHeight20',
  textColor: 'colorText',
};

export {Text};
