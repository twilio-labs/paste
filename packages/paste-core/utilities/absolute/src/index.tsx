import styled from '@emotion/styled';
import {top, right, bottom, left, zIndex, TopProps, BottomProps, LeftProps, RightProps, styleFn} from 'styled-system';
import {Box, BoxProps} from '@twilio-paste/box';
import {ThemeShape} from '@twilio-paste/theme-tokens';

const LEFT_OPTIONS = [
  'fill',
  'left',
  'left_top',
  'left_bottom',
  'left_fill',
  'top_left',
  'top_fill',
  'bottom_left',
  'bottom_fill',
];
const RIGHT_OPTIONS = [
  'fill',
  'right',
  'right_top',
  'right_bottom',
  'right_fill',
  'top_right',
  'top_fill',
  'bottom_right',
  'bottom_fill',
];
const TOP_OPTIONS = [
  'fill',
  'top',
  'top_right',
  'top_left',
  'top_fill',
  'right_top',
  'right_fill',
  'left_top',
  'left_fill',
];
const BOTTOM_OPTIONS = [
  'fill',
  'bottom',
  'bottom_right',
  'bottom_left',
  'bottom_fill',
  'right_bottom',
  'right_fill',
  'left_bottom',
  'left_fill',
];

export type Presets =
  | 'fill'
  | 'top'
  | 'top_left'
  | 'top_right'
  | 'top_fill'
  | 'bottom'
  | 'bottom_left'
  | 'bottom_right'
  | 'bottom_fill'
  | 'left'
  | 'left_top'
  | 'left_bottom'
  | 'left_fill'
  | 'right'
  | 'right_top'
  | 'right_bottom'
  | 'right_fill';

export interface ZIndexProps {
  zIndex?: keyof ThemeShape['zIndices'];
}

export interface AbsoluteProps extends TopProps, BottomProps, LeftProps, RightProps, ZIndexProps, BoxProps {
  preset?: Presets;
}

interface PositionStyles {
  top?: 0;
  right?: 0;
  bottom?: 0;
  left?: 0;
}

const preset: styleFn = ({preset: _preset}) => {
  const positionStyles: PositionStyles = {};

  if (_preset == null) {
    return positionStyles;
  }

  if (LEFT_OPTIONS.includes(_preset)) {
    positionStyles.left = 0;
  }
  if (RIGHT_OPTIONS.includes(_preset)) {
    positionStyles.right = 0;
  }
  if (TOP_OPTIONS.includes(_preset)) {
    positionStyles.top = 0;
  }
  if (BOTTOM_OPTIONS.includes(_preset)) {
    positionStyles.bottom = 0;
  }
  return positionStyles;
};

const Absolute = styled(Box)<AbsoluteProps>(preset, zIndex, top, right, bottom, left, {
  position: 'absolute',
});

Absolute.defaultProps = {
  preset: 'fill',
};

export {Absolute};
