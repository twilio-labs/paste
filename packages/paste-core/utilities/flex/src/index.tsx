import * as React from 'react';
import * as PropTypes from 'prop-types';
import {ResponsiveValue} from 'styled-system';
import {Box} from '@twilio-paste/box';
import {FlexboxProps} from '@twilio-paste/style-props';

type DisplayOptions = 'flex' | 'inline-flex';
export type Display = ResponsiveValue<DisplayOptions>;
type VerticalAlignOptions = 'top' | 'center' | 'bottom' | 'stretch';
export type VerticalAlign = ResponsiveValue<VerticalAlignOptions>;
type HorizontalAlignOptions = 'left' | 'center' | 'right' | 'around' | 'between';
export type HorizontalAlign = ResponsiveValue<HorizontalAlignOptions>;
type VerticalOptions = boolean;
export type Vertical = ResponsiveValue<VerticalOptions>;
type GrowOptions = boolean | number;
export type Grow = ResponsiveValue<GrowOptions>;
type ShrinkOptions = boolean | number;
export type Shrink = ResponsiveValue<ShrinkOptions>;
type BasisOptions = string | number;
export type Basis = ResponsiveValue<BasisOptions>;
type WrapOptions = boolean;
export type Wrap = ResponsiveValue<WrapOptions>;

export interface FlexProps {
  display?: Display;
  vertical?: Vertical;
  vAlignContent?: VerticalAlign;
  hAlignContent?: HorizontalAlign;
  grow?: Grow;
  shrink?: Shrink;
  basis?: Basis;
  wrap?: Wrap;
}

export const getGrow = ({grow}: FlexProps): {} => {
  if (Array.isArray(grow)) {
    return (grow as GrowOptions[]).map((value: GrowOptions) => {
      return Number(value);
    });
  }

  if (typeof grow === 'number') {
    return grow;
  }

  if (grow) {
    return 1;
  }

  return 0;
};

export const getShrink = ({shrink, basis}: FlexProps): {} => {
  if (Array.isArray(shrink)) {
    return (shrink as ShrinkOptions[]).map((value: ShrinkOptions) => {
      return Number(value);
    });
  }

  if (typeof shrink === 'number') {
    return shrink;
  }

  if (typeof shrink === 'boolean') {
    return shrink ? 1 : 0;
  }

  if (basis && basis !== 'auto') {
    return 0;
  }

  return 1;
};

export const getSuffix = (item: Basis): {} => {
  const suffix = typeof item === 'number' || String(parseInt(item as string, 10)) === item ? 'px' : '';
  return item + suffix;
};

export const getBasis = ({basis}: FlexProps): {} => {
  if (Array.isArray(basis)) {
    return (basis as BasisOptions[]).map((value: BasisOptions) => {
      return getSuffix(value);
    });
  }

  if (basis) {
    return getSuffix(basis);
  }

  return 'auto';
};

export const getVertical = ({vertical}: FlexProps): {} => {
  if (Array.isArray(vertical)) {
    return (vertical as VerticalOptions[]).map((value: VerticalOptions) => {
      if (typeof value === 'boolean') {
        return value === true ? 'column' : 'row';
      }
      return value;
    });
  }

  if (vertical) {
    return 'column';
  }

  return 'row';
};

export const getWrap = ({wrap}: FlexProps): {} => {
  if (Array.isArray(wrap)) {
    return (wrap as WrapOptions[]).map((value: WrapOptions) => {
      if (typeof value === 'boolean') {
        return value === true ? 'wrap' : 'nowrap';
      }
      return value;
    });
  }

  if (wrap) {
    return 'wrap';
  }

  return 'nowrap';
};

const RemapedVerticalAlignments = {
  top: 'flex-start',
  center: 'center',
  bottom: 'flex-end',
  stretch: 'stretch',
};

export const vAlignToProps = ({vAlignContent}: FlexProps): {} => {
  if (Array.isArray(vAlignContent)) {
    return (vAlignContent as VerticalAlignOptions[]).map(value => RemapedVerticalAlignments[value]);
  }

  if (vAlignContent) {
    return RemapedVerticalAlignments[vAlignContent as VerticalAlignOptions];
  }

  return 'flex-start';
};

const RemapedHorizontalAlignments = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
  around: 'space-around',
  between: 'space-between',
};

export const hAlignToProps = ({hAlignContent}: FlexProps): {} => {
  if (Array.isArray(hAlignContent)) {
    return (hAlignContent as HorizontalAlignOptions[]).map(value => RemapedHorizontalAlignments[value]);
  }

  if (hAlignContent) {
    return RemapedHorizontalAlignments[hAlignContent as HorizontalAlignOptions];
  }

  return 'flex-start';
};

const getFlexStyles = (props: FlexProps): FlexboxProps => {
  const styles: FlexboxProps = {
    justifyContent: props.vertical ? vAlignToProps(props) : hAlignToProps(props),
    alignItems: props.vertical ? hAlignToProps(props) : vAlignToProps(props),
  };

  if (props.grow || props.shrink || props.basis) {
    styles.flexGrow = getGrow(props);
    styles.flexShrink = getShrink(props);
    styles.flexBasis = getBasis(props);
  }

  if (props.vertical) {
    styles.flexDirection = getVertical(props);
  }

  if (props.wrap) {
    styles.flexWrap = getWrap(props);
  }

  return styles;
};

const Flex: React.FC<FlexProps> = props => {
  const FlexStyles = React.useMemo(() => getFlexStyles(props), [props]);
  return (
    <Box {...FlexStyles} display={props.display}>
      {props.children}
    </Box>
  );
};

Flex.propTypes = {
  display: PropTypes.oneOfType([
    PropTypes.oneOf(['flex', 'inline-flex']),
    PropTypes.arrayOf(PropTypes.oneOf(['flex', 'inline-flex'])),
  ] as any),
  vertical: PropTypes.oneOfType([
    PropTypes.oneOfType([PropTypes.bool]),
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.bool])),
  ] as any),
  vAlignContent: PropTypes.oneOfType([
    PropTypes.oneOf(['top', 'center', 'bottom', 'stretch']),
    PropTypes.arrayOf(PropTypes.oneOf(['top', 'center', 'bottom', 'stretch'])),
  ] as any),
  hAlignContent: PropTypes.oneOfType([
    PropTypes.oneOf(['left', 'center', 'right', 'around', 'between']),
    PropTypes.arrayOf(PropTypes.oneOf(['left', 'center', 'right', 'around', 'between'])),
  ] as any),
  grow: PropTypes.oneOfType([
    PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.number])),
  ] as any),
  shrink: PropTypes.oneOfType([
    PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.number])),
  ] as any),
  basis: PropTypes.oneOfType([
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  ] as any),
  wrap: PropTypes.oneOfType([
    PropTypes.oneOfType([PropTypes.bool]),
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.bool])),
  ] as any),
};

Flex.defaultProps = {
  display: 'flex',
};

Flex.displayName = 'Flex';
export {Flex};
