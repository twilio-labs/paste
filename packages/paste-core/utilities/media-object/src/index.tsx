import * as React from 'react';
import {Box, BoxProps} from '@twilio-paste/box';
import {Space} from '@twilio-paste/style-props';

export interface MediaObjectProps extends Pick<BoxProps, 'as'> {
  verticalAlign?: 'center' | 'top';
}

const MediaObject: React.FC<MediaObjectProps> = ({as = 'span', children, verticalAlign = 'top'}) => {
  return (
    <Box as={as} display="flex" alignItems={verticalAlign === 'top' ? 'flex-start' : 'center'}>
      {children}
    </Box>
  );
};
MediaObject.displayName = 'MediaObject';

export interface MediaFigureProps extends Pick<BoxProps, 'as'> {
  align?: 'start' | 'end';
  spacing?: Space;
}
const MediaFigure: React.FC<MediaFigureProps> = ({as = 'span', children, align, spacing}) => {
  return (
    <Box
      as={as}
      display="flex"
      flexShrink={0}
      marginLeft={align === 'end' ? spacing : undefined}
      marginRight={align === 'start' ? spacing : undefined}
    >
      {children}
    </Box>
  );
};
MediaFigure.displayName = 'MediaFigure';
MediaFigure.defaultProps = {
  align: 'start',
  spacing: 'space0',
};

export type MediaBodyProps = Pick<BoxProps, 'as'>;
const MediaBody: React.FC<MediaBodyProps> = ({as = 'span', children}) => {
  return (
    <Box as={as} flex={1}>
      {children}
    </Box>
  );
};
MediaBody.displayName = 'MediaBody';

export {MediaObject, MediaFigure, MediaBody};
