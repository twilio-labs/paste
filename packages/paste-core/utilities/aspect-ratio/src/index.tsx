import * as React from 'react';
import styled from '@emotion/styled';
import {Absolute} from '@twilio-paste/absolute';

interface AspectRatioProps {
  ratio: string;
  children: NonNullable<React.ReactNode>;
}

const RATIO_REGEX = /^(\d+:\d*)$/;

const isCorrectPattern = (ratio: string): boolean => RATIO_REGEX.test(ratio);

const handlePropValidation = ({ratio}: AspectRatioProps): void => {
  const hasRatio = ratio != null && ratio !== '';

  if (!hasRatio) {
    throw new Error(`[Paste: AspectRatio] Missing 'ratio' prop.`);
  }

  if (!isCorrectPattern(ratio)) {
    throw new Error(`[Paste: AspectRatio] 'ratio' is invalid. Use a colon separated number pattern (4:3).`);
  }
};

const AspectRatioContainer = styled.div`
  position: relative;
`;

const AspectRatio: React.FC<AspectRatioProps> = props => {
  handlePropValidation(props);

  const aspectArray = props.ratio.split(':').map(Number);
  const aspectPercent = (aspectArray[1] / aspectArray[0]) * 100;

  return (
    <AspectRatioContainer style={{paddingBottom: `${aspectPercent}%`}}>
      <Absolute>{props.children}</Absolute>
    </AspectRatioContainer>
  );
};

AspectRatio.defaultProps = {
  ratio: '4:3',
};

export {AspectRatio};
