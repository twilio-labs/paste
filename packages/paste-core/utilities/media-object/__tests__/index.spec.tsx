import * as React from 'react';
import renderer from 'react-test-renderer';
import {ReactWrapper, mount} from 'enzyme';
import {Theme} from '@twilio-paste/theme';
import {MediaObject, MediaFigure, MediaBody} from '../src';

describe('MediaObject', () => {
  it('should render', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaObject>default</MediaObject>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with center aligned children', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaObject verticalAlign="center">center align</MediaObject>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as another HTML element', (): void => {
    const wrapper: ReactWrapper = mount(<MediaObject as="section">background single</MediaObject>);
    expect(wrapper.exists('section')).toEqual(true);

    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaObject as="section">background single</MediaObject>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render bottom margin', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaObject marginBottom="space40">background single</MediaObject>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render top margin', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaObject marginTop="space40">background single</MediaObject>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MediaFigure', () => {
  it('should render', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaFigure>default</MediaFigure>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as any HTML element', (): void => {
    const wrapper: ReactWrapper = mount(<MediaFigure as="a">default</MediaFigure>);
    expect(wrapper.exists('a')).toEqual(true);

    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaFigure as="a">default</MediaFigure>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with spacing on the right for default alignment', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaFigure spacing="space40">center align</MediaFigure>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with spacing on the left for end alignment', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaFigure align="end" spacing="space40">
            center align
          </MediaFigure>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MediaBody', () => {
  it('should render', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaBody>default</MediaBody>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as any HTML element', (): void => {
    const wrapper: ReactWrapper = mount(<MediaBody as="article">default</MediaBody>);
    expect(wrapper.exists('article')).toEqual(true);

    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <MediaBody as="article">default</MediaBody>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
