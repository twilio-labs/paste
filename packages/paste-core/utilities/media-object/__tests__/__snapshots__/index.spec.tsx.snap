// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaBody should render 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
  >
    default
  </span>
</div>
`;

exports[`MediaBody should render as any HTML element 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-1"
>
  <article
    className="emotion-0"
  >
    default
  </article>
</div>
`;

exports[`MediaFigure should render 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
    display="flex"
  >
    default
  </span>
</div>
`;

exports[`MediaFigure should render as any HTML element 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0;
}

<div
  className="emotion-1"
>
  <a
    className="emotion-0"
    display="flex"
  >
    default
  </a>
</div>
`;

exports[`MediaFigure should render with spacing on the left for end alignment 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 0.75rem;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
    display="flex"
  >
    center align
  </span>
</div>
`;

exports[`MediaFigure should render with spacing on the right for default alignment 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 0.75rem;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
    display="flex"
  >
    center align
  </span>
</div>
`;

exports[`MediaObject should render 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
    display="flex"
  >
    default
  </span>
</div>
`;

exports[`MediaObject should render as another HTML element 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="emotion-1"
>
  <section
    className="emotion-0"
    display="flex"
  >
    background single
  </section>
</div>
`;

exports[`MediaObject should render with center aligned children 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
    display="flex"
  >
    center align
  </span>
</div>
`;
